// @atlaskit/form/dist/esm/fieldset.js
/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import { useGlobalTheme } from '@atlaskit/theme/components';
import { fontFamily as getFontFamily, gridSize as getGridSize } from '@atlaskit/theme/constants';
import { h200 } from '@atlaskit/theme/typography';
var fontFamily = getFontFamily();
var gridSize = getGridSize();
var fieldsetLabelStyles = css({
  display: 'inline-block',
  marginTop: 0,
  marginBottom: 0,
  fontFamily: "".concat(fontFamily)
});
var fieldSetStyles = css({
  marginTop: "".concat(gridSize, "px")
}); // eslint-disable-next-line @repo/internal/react/consistent-css-prop-usage

var lightH200Styles = css(h200({
  theme: {
    mode: 'light'
  }
})); // eslint-disable-next-line @repo/internal/react/consistent-css-prop-usage

var darkH200Styles = css(h200({
  theme: {
    mode: 'dark'
  }
}));

var FieldsetLabel = function FieldsetLabel(_ref) {
  var children = _ref.children;

  var _useGlobalTheme = useGlobalTheme(),
      mode = _useGlobalTheme.mode;

  return jsx("label", {
    css: [mode === 'light' ? lightH200Styles : darkH200Styles, fieldsetLabelStyles]
  }, children);
};
/**
 * __Fieldset__
 *
 * A fieldset groups a number of fields together. For example, when multiple CheckboxFields share the same name,
 * a fieldset can be used to group them together. This makes the form more accessible.
 *
 * - [Examples](https://atlaskit.atlassian.com/packages/design-system/form/docs/fields)
 * - [Code](https://atlaskit.atlassian.com/packages/design-system/form/docs/fields)
 * - [Usage](https://atlaskit.atlassian.com/packages/design-system/form/docs/fields)
 */


var Fieldset = function Fieldset(_ref2) {
  var children = _ref2.children,
      legend = _ref2.legend;
  return jsx("fieldset", {
    css: fieldSetStyles
  }, legend && jsx("legend", null, jsx(FieldsetLabel, null, legend)), children);
};

export default Fieldset;
