// @atlaskit/form/dist/cjs/fieldset.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _core = require("@emotion/core");

var _components = require("@atlaskit/theme/components");

var _constants = require("@atlaskit/theme/constants");

var _typography = require("@atlaskit/theme/typography");

/** @jsx jsx */
var fontFamily = (0, _constants.fontFamily)();
var gridSize = (0, _constants.gridSize)();
var fieldsetLabelStyles = (0, _core.css)({
  display: 'inline-block',
  marginTop: 0,
  marginBottom: 0,
  fontFamily: "".concat(fontFamily)
});
var fieldSetStyles = (0, _core.css)({
  marginTop: "".concat(gridSize, "px")
}); // eslint-disable-next-line @repo/internal/react/consistent-css-prop-usage

var lightH200Styles = (0, _core.css)((0, _typography.h200)({
  theme: {
    mode: 'light'
  }
})); // eslint-disable-next-line @repo/internal/react/consistent-css-prop-usage

var darkH200Styles = (0, _core.css)((0, _typography.h200)({
  theme: {
    mode: 'dark'
  }
}));

var FieldsetLabel = function FieldsetLabel(_ref) {
  var children = _ref.children;

  var _useGlobalTheme = (0, _components.useGlobalTheme)(),
      mode = _useGlobalTheme.mode;

  return (0, _core.jsx)("label", {
    css: [mode === 'light' ? lightH200Styles : darkH200Styles, fieldsetLabelStyles]
  }, children);
};
/**
 * __Fieldset__
 *
 * A fieldset groups a number of fields together. For example, when multiple CheckboxFields share the same name,
 * a fieldset can be used to group them together. This makes the form more accessible.
 *
 * - [Examples](https://atlaskit.atlassian.com/packages/design-system/form/docs/fields)
 * - [Code](https://atlaskit.atlassian.com/packages/design-system/form/docs/fields)
 * - [Usage](https://atlaskit.atlassian.com/packages/design-system/form/docs/fields)
 */


var Fieldset = function Fieldset(_ref2) {
  var children = _ref2.children,
      legend = _ref2.legend;
  return (0, _core.jsx)("fieldset", {
    css: fieldSetStyles
  }, legend && (0, _core.jsx)("legend", null, (0, _core.jsx)(FieldsetLabel, null, legend)), children);
};

var _default = Fieldset;
exports.default = _default;
